File: java/Endpoint.java
package irresponsible.crouton;

import clojure.lang.IPersistentVector;
import clojure.lang.PersistentArrayMap;
import clojure.lang.ITransientMap;
import clojure.lang.Keyword;

public class Endpoint extends AEndpoint {
  private final Object ret = Keyword.intern("crouton","route");
  private final Object val;

  public Endpoint(Object v) {
    if (null == v)
      throw new IllegalArgumentException("value must not be nil");
    val = v;
  }
  public final Object match(IPersistentVector pieces, ITransientMap places) {
    if (0 != pieces.length()) return null;
    if (0 == places.count()) return PersistentArrayMap.EMPTY.assoc(ret,val);
    return 0 == pieces.count() ? places.assoc(ret,val).persistent() : null;
  }
}

File: kotlin/src/Endpoint.kt
package irresponsible.crouton

import clojure.lang.IPersistentVector
import clojure.lang.PersistentArrayMap
import clojure.lang.IPersistentMap
import clojure.lang.Keyword

class Endpoint(private val value: Any) : AEndpoint() {
    private val ret = Keyword.intern("crouton", "route")
    override fun match(pieces: IPersistentVector, places: IPersistentMap): Any? {
        if (0 != pieces.length()) return null
        if (0 == places.count()) return PersistentArrayMap.EMPTY.assoc(ret, value)
        return if (0 == pieces.count()) places.assoc(ret, value) else null
    }
}
